### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by iest**
We now have unit tests!! :tada: :sparkles: 

To wrap up this PR:

- [x] Spec for `Btn`
- [x] Spec for `Checkbox`
- [x] Spec for `Toggle`
- [x] Get tests running on circleci
- [x] Add howto to readme

### CODE REVIEW

**Comment by iest** [comment]
Wondering if it would be useful to publish the spec outputs somewhere. Very nice way to understand what you can expect from a component:

```
  Process environment for tests
    ✔ Should be development for React console warnings
  Btn
    ✔ Should render a button
    ✔ Should output an anchor when passed an href
    ✔ Passes through the className to the actual DOM element
    ✔ Doesn't pass through all props it's given
    ✔ Corrently applies the className for the type
    ✔ Corrently applies the className for the variant
    ✔ Applies the correct CSS class when the fullWidth prop is passed
    ✔ Disables correctly as a button
    ✔ Can't be disabled if it's an anchor with an href
    ✔ Calls the onClick function it's passed
  Checkbox
    ✔ Should render an input, label, and icon within a wrapper
    ✔ Calls onChange when changed
    ✔ Has a label with a "for" attribute that matches the input's "id"
    ✔ Has a unique id so there's no mismatching "for"s and "ids"
    ✔ Is checked when the checked prop is true
    ✔ Is not checked when the checked prop is false
    ✔ Accepts a name attribute and applies it to the input
    ✔ Passes through the className
  Toggle
    ✔ Should render an input inside a label inside a wrapper
    ✔ Calls onChange when changed
    ✔ Is on when the value is true
    ✔ Is off when the value is false
    ✔ Passes through the className
```

**Comment by tomgatzgates** [comment]
a collapsable box below the style guide component?

**Comment by iest** [comment]
Good idea. No idea how we'd actually get the output in there though.

File: components/__tests__/Btn-test.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  });
+
+  it(`Doesn't pass through all props it's given`, () => {
+    const node = findDOMNode(renderIntoDocument(
+      <Btn data-someRandoAttribute="derp"/>
+    ));
+    assert.equal(node.getAttribute('data-someRandoAttribute'), null);
+  });
+
+  it('Corrently applies the className for the type', () => {
```
Comment (tomgatzgates) [inline components/__tests__/Btn-test.js]: spelling "Corrently"

**Pull request merged**